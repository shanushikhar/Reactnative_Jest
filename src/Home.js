import React, { Component } from 'react'
import { Button, Text, TextInput, View } from 'react-native'
import Users from './users'

export default class Home extends Component {
    state = {
        data: '',
        count: 0
    }

    componentDidMount() {
        Users.all().then(data => { })
        this.setState({
            data: 'data added...'
        })
    }

    change(x) {
        //  return x * 5  

        this.setState({ data: x * 5 })
    }

    refreshData = () => {
        console.log("download data from server placeholder");
        this.setState(
            (prevState, props) => {
                return { count: prevState.count + 1 };
            }
        );
    }

    render() {
        return (
            <View>
                <TextInput
                    testID={'username'}
                    onChangeText={(text) => this.setState(text)}
                    style={{ backgroundColor: 'grey', marginBottom: 15 }}
                    placeholder='Enter username'
                    placeholderTextColor='red'
                />
                <Text>hey {this.props.data}</Text>
                <TextInput
                    testID={'password'}
                    onChangeText={(text) => this.setState(text)}
                    style={{ backgroundColor: 'grey', marginBottom: 15 }}
                    placeholder='Enter password'
                    placeholderTextColor='red'
                />
                <Button
                    id="refreshButton"
                    onPress={this.refreshData}
                    title="Refresh"
                    color="#000000"
                    accesibilityLabel="Refresh AQI data"
                    testID="refreshButton"
                />
            </View>
        )
    }
}
