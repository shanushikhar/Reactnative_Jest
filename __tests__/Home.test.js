import React from 'react'
import renderer from 'react-test-renderer'
import fetch from 'node-fetch'

import Home from '../src/Home'
import Users from '../src/users'


test('Snapshot testing', () => {
    const snap = renderer.create(<Home />).toJSON()

    expect(snap).toMatchSnapshot()
})

test('method testing', () => {
    const Homedata = renderer.create(<Home />).getInstance()
    // console.log(Homedata)

    // expect(Homedata.change(5)).toEqual(25)

    Homedata.change(2)
    expect(Homedata.state.data).toEqual(10)
})

let findElement = function (tree, elemet) {
    // console.warn(tree.children)
    // for (let node in tree.children) {
    //     console.log(tree.children[node].type)
    // }

    let res = ''
    for (let x of tree.children) {
        if (x.props.testID == elemet) {
            res = true
        }
        return res
    }
}

test('element testing', () => {
    const Homedata = renderer.create(<Home />).toJSON()
    // console.log(Homedata)

    expect(findElement(Homedata, 'username')).toBeDefined()

})


test('API testing', async function () {

    // global.fetch = jest.fn().mockImplementation(() => {
    //     let p = new Promise((resolve, reject) => {
    //         resolve({
    //             json: function () {
    //                 return { ID: 1 }
    //             }
    //         })
    //     })
    //     return p
    // })

    const response = await Users.all()
  //  console.log('datas.....', response)

    expect(response.title).toBe('The Basics - Networking')
})