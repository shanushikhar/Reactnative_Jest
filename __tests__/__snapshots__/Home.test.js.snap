// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter username"
    placeholderTextColor="red"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "grey",
        "marginBottom": 15,
      }
    }
    testID="username"
    underlineColorAndroid="transparent"
  />
  <Text>
    hey 
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter password"
    placeholderTextColor="red"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "grey",
        "marginBottom": 15,
      }
    }
    testID="password"
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="refreshButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#000000",
            },
          ]
        }
      >
        Refresh
      </Text>
    </View>
  </View>
</View>
`;
